include/ensmallen_bits/callbacks/query_front.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file query_front.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * Implementation of the query front callback function.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_CALLBACKS_QUERY_FRONT_HPP
+#define ENSMALLEN_CALLBACKS_QUERY_FRONT_HPP
+
+namespace ens {
+
+/**
+ * Query the current Pareto Front after every GenerationalStepTaken callback function.
+ */
+class QueryFront
+{
+ public:
+  /**
+   * Set up the query front callback class with the specified inputs.
+   *
+   * @param queryRate The frequency at which the Pareto Front is queried.
+   * @param paretoFrontArray A reference to a vector of cube to store the queried fronts.
+   * @param genCounter Internal clock to count generations passed.
```

Comment:
Looks like I can't pass `genCounter`.

Reply:
Yes, this is an internal private counter meant to be in sync with ```generations++``` inside the main loop of optimization.

Reply:
Think of it similar to ```steps``` in ```early_stop_at_min_loss.hpp```

Reply:
Ah, as per convention, I should remove the doc from there. Sure!

Reply:
Yes, let's remove it from the doc.

Reply:
Already done 👍